src/sidebar/components/MarkdownView.tsx

```diff
@@ -22,14 +44,49 @@ export default function MarkdownView({
   markdown,
   classes,
   style,
-  mentions,
+  mentions = [],
+  mentionsEnabled = false,
+  /* istanbul ignore next - test seam */
+  setTimeout_ = setTimeout,
+  /* istanbul ignore next - test seam */
+  clearTimeout_ = clearTimeout,
 }: MarkdownViewProps) {
   const html = useMemo(
     () => (markdown ? renderMathAndMarkdown(markdown) : ''),
     [markdown],
   );
   const content = useRef<HTMLDivElement | null>(null);
 
+  const mentionsPopoverAnchorRef = useRef<HTMLElement | null>(null);
+  const elementToMentionMap = useRef(
+    new Map<HTMLElement, Mention | InvalidUsername>(),
+  );
+  const [popoverContent, doSetPopoverContent] = useState<PopoverContent>(null);
+  const popoverContentTimeout = useRef<ReturnType<typeof setTimeout> | null>();
+  const setPopoverContent = useCallback(
```

Comment:
Maybe name this `setPopoverContentAfterDelay` and then the immediate `doSetPopoverContent` can have its normal name.